python/cudf/cudf/core/udf/groupby_utils.py

```diff
@@ -211,18 +182,52 @@ def _can_be_jitted(frame, func, args):
 
     if any(col.has_nulls() for col in frame._columns):
         return False
-    np_field_types = np.dtype(
-        list(
-            _supported_dtypes_from_frame(
-                frame, supported_types=SUPPORTED_GROUPBY_NUMPY_TYPES
-            ).items()
-        )
-    )
-    dataframe_group_type = _get_frame_groupby_type(
-        np_field_types, frame.index.dtype
-    )
+    kr = GroupByApplyKernel(frame, func, args)
     try:
-        _get_udf_return_type(dataframe_group_type, func, args)
+        kr._get_udf_return_type()
         return True
     except (UDFError, TypingError):
         return False
+
+
+class GroupByApplyKernel(ApplyKernelBase):
+    """
+    Class representing a kernel that computes the result of
+    a GroupBy.apply operation. Expects that the user passed
+    a function that operates on a single group of the data,
+    for example
+
+    def f(group):
+        return group['x'].sum() + group['y'].sum()
+    """
+
+    def f(x):
+        return x + 1
```

Comment:
Do we need this? I wasn't sure if it was being analyzed somewhere by numba to determine a signature. If so, it needs a comment. Otherwise it looks superfluous.